R/manual_layer_node.R

```diff
@@ -354,47 +357,58 @@ Node <- R6::R6Class(
 ##'
 ##' @family {manual-layer functions}
 ##' @export
-compute <- function(object, namespace=NULL, timeout_seconds=NULL, verbose=FALSE, force_all_local=FALSE) 0
-setMethod("compute", signature(object = "Node"), function(object, namespace=NULL, timeout_seconds=NULL,
+compute <- function(delayed_function, namespace=NULL, timeout_seconds=NULL, verbose=FALSE, force_all_local=FALSE) 0
```

Comment:
Rename `object` to `delayed_function` for consistency with the below.

Reply:
I still think a trailing `0` is hard-read, non-standard and would prefer `{}` there.

Reply:
Oops, sorry @eddelbuettel about the 0. I had fixed an instance on another PR but missed that here -- thank you!

My use of `0` was copypasta from a reference I found online; thank you for providing a better one!

My goal is indeed simply to create a generic! In addition to `x$foo()` I want to make `foo(x)` for important things like `compute` -- that's my only goal. The syntax I used in doing so was to avoid errors: if I omit the line

```
compute <- function(delayed_function, namespace=NULL, timeout_seconds=NULL, verbose=FALSE, force_all_local=FALSE) {}
```

then I get the following error on `devtools::load_all()`:

```
Loading tiledbcloud
in method for ‘show’ with signature ‘object="DAG"’: no definition for class “DAG”
Error in setMethod("compute", signature(delayed_function = "Node"), function(delayed_function,  :
  no existing definition for function ‘compute’
```

Thanks for your links on dispatching -- I'll study them and learn from them and try to find a more elegant syntax for generics which doesn't generate syntax errors -- thanks again!